We want to break out of this loop after the timeout expires.

diff -uNr Linux-PAM-0.75/modules/pam_unix/pam_unix_passwd.c Linux-PAM-0.75/modules/pam_unix/pam_unix_passwd.c
--- Linux-PAM-0.75/modules/pam_unix/pam_unix_passwd.c	Sun Feb 11 01:33:53 2001
+++ Linux-PAM-0.75/modules/pam_unix/pam_unix_passwd.c	Tue Jul 10 00:14:23 2001
@@ -666,6 +666,7 @@
 	i=0;
 	while((retval = lckpwdf()) != 0 && i < 100) {
 		usleep(1000);
+		i++;
 	}
 	if(retval != 0) {
 		return PAM_AUTHTOK_LOCK_BUSY;
